packages/dashboard-frontend/src/services/workspace-client/devworkspace/DevWorkspaceDefaultPluginsHandler.ts

```diff
@@ -0,0 +1,125 @@
+  public async handle(workspace: devfileApi.DevWorkspace, editor: string): Promise<void> {
+    const defaultPlugins = await ServerConfigApi.getDefaultPlugins(editor);
+    this.handleUriPlugins(workspace, defaultPlugins);
+    this.patchWorkspaceComponents(workspace);
+  }
+
+  private async handleUriPlugins(workspace: devfileApi.DevWorkspace, defaultPlugins: string[]) {
+    const defaultUriPlugins = new Set(defaultPlugins.filter(this.isUri));
+
+    this.removeOldDefaultUriPlugins(workspace, defaultUriPlugins);
+
+    defaultUriPlugins.forEach(plugin => {
+      const name = createHash('MD5').update(plugin).digest('hex').substring(0, 20).toLowerCase();
+      this.addDefaultPluginByUri(workspace, 'default-' + name, plugin);
+    });
+  }
+
+  private isUri(str: string) {
+    try {
+      new URL(str);
+      return true;
+    } catch (err) {
+      return false;
+    }
+  }
+
+  /**
+   * Checks if there are default plugins in the workspace that are not
+   * specified in defaultUriPlugins. If such plugins are found, this function
+   * removes them.
+   * @param workspace A devworkspace to remove old default plugins for
+   * @param defaultUriPlugins The set of current default plugins
+   */
+  private removeOldDefaultUriPlugins(
+    workspace: devfileApi.DevWorkspace,
+    defaultUriPlugins: Set<string>,
+  ) {
+    if (!workspace.spec.template.components) {
+      return;
+    }
+    const components = workspace.spec.template.components.filter(component => {
+      if (!this.isDefaultPluginComponent(component) || !component.plugin?.uri) {
+        // component is not a default uri plugin, keep component.
+        return true;
+      }
+      return defaultUriPlugins.has(component.plugin.uri);
+    });
+    workspace.spec.template.components = components;
+  }
+
+  /**
+   * Returns true if component is a plugin managed by this class
+   * @param component The component to check
+   * @returns true if component is a plugin managed by this class
+   */
+  private isDefaultPluginComponent(component: V1alpha2DevWorkspaceSpecTemplateComponents): boolean {
+    return component.attributes && component.attributes[DEFAULT_PLUGIN_ATTRIBUTE]
+      ? component.attributes[DEFAULT_PLUGIN_ATTRIBUTE].toString() === 'true'
+      : false;
+  }
+
+  /**
+   * Add a default plugin to the workspace by uri
+   * @param workspace A devworkspace
+   * @param pluginName The name of the plugin
+   * @param pluginUri The uri of the plugin
+   */
+  private addDefaultPluginByUri(
+    workspace: devfileApi.DevWorkspace,
+    pluginName: string,
+    pluginUri: string,
+  ) {
+    if (!workspace.spec.template.components) {
+      workspace.spec.template.components = [];
+    }
+
+    if (workspace.spec.template.components.find(component => component.name === pluginName)) {
+      // plugin already exists
+      return;
+    }
+
+    workspace.spec.template.components.push({
+      name: pluginName,
+      attributes: { [DEFAULT_PLUGIN_ATTRIBUTE]: true },
```

Comment:
An attribute is being set here (`default-plugin`), in order to distinguish between non-default and default plugins in  `spec.template.components`:
![image](https://user-images.githubusercontent.com/83611742/145901191-5fd6564a-4bf3-4714-ab68-9503b363f0c1.png)